workflow/Snakefile

```diff
@@ -86,29 +58,82 @@ rule map_reads:
     minimap2 -t {resources.cpus_per_task} -ax map-ont -p {params.psec} -N {params.msec} {params.opts} {input.index} {input.fastq} > {output} 2> {log}
     """
 
+#QC and metadata with NanoPlot
+if config["summary"] == "None":
+    rule plot_samples:
+        input:
+            fastq = lambda wildcards: all_samples[wildcards.sample]
+        output:
+            directory("NanoPlot")
+        log:
+            "logs/NanoPlot/{sample}.log"
+        resources:
+            ## max of 39 for our SLURM partition
+            cpus_per_task= max(8,39) #problem with max(len(input.fastq),39)
+        conda:
+            "envs/env.yml"
+        shell: 
+            "mkdir {output}/{wildcards.sample}; "
+            "NanoPlot -t {resources.cpus_per_task} --tsv_stats "
+            "--fastq {input.fastq} -o {output}/{wildcards.sample} 2> {log}"
+else:
+    rule plot_samples:
+        input:
+            summary = config["summary"]
+        output:
+            directory("NanoPlot")
+        log:
+            "logs/NanoPlot/NanoPlot.log"
+        conda:
+            "envs/env.yml"
+        shell: 
+            "mkdir {output}/{wildcards.sample}; "
+            "NanoPlot -t {resources.cpus_per_task} --barcoded --tsv_stats "
+            "--summary {input.summary} -o {output} 2> {log}"
+
+rule compress_nplot:
+    input:
+        samples=rules.plot_samples.output
+    output:
+        "NanoPlot.tar.gz"
+    log:
+        "logs/NanoPlot/compress.log"
+    shell:
+        "tar zcvf {input}.tar.gz {input} 2> {log}"
+
```

Comment:
You will hate me for this - but a good night's sleep lets me think: What would I as a user like to see? Isn't it both, the per-sample analysis and the broad summary?